aws/securityhub.go

```diff
@@ -0,0 +1,97 @@
+package aws
+
+import (
+	"strings"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/hashicorp/go-multierror"
+)
+
+func getAllSecurityHubInsights(session *session.Session, region string) ([]*string, error) {
+	svc := securityhub.New(session)
+	result, err := svc.GetInsights(&securityhub.GetInsightsInput{})
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	var names []*string
+	names = append(names, awsgo.String("disable-securityhub"))
+	for _, insight := range result.Insights {
+		names = append(names, insight.Name)
+	}
+
+	return names, nil
+}
+
+func deleteSecurityHubInsight(svc *securityhub.SecurityHub, insightArn *string) error {
+	_, err := svc.DeleteInsight(&securityhub.DeleteInsightInput{
+		InsightArn: insightArn,
+	})
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	return nil
+}
+
+func nukeAllSecurityHubInsights(session *session.Session, insights []*string) error {
+	if len(insights) == 0 {
+		logging.Logger.Info("No Security Hub Insights to nuke")
+		return nil
+	}
+
+	logging.Logger.Info("Deleting all Security Hub Insights")
+
+	deletedinsights := 0
+	svc := securityhub.New(session)
+	multiErr := new(multierror.Error)
+
+	for _, name := range insights {
+		if *name == "disable-securityhub" {
```

Comment:
Ah for consistency's sake - we should use the aws SDK method for dereferencing a pointer.